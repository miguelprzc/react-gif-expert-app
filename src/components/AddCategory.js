import React, { useState } from 'react'
import { PropTypes } from 'prop-types';

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim()) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text"
        placeholder="What are you looking for?"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
