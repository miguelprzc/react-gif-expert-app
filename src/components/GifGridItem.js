import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
  return (
    <figure className=" card animate__animated animate__fadeInUp">
      <img src={url} alt={title} />
      <figcaption>{title}</figcaption>
    </figure>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
