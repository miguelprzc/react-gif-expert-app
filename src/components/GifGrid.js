import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

  const {data: gifs, loading} = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeInUp">{category}</h3>
      <div className="card-grid">
        {
          loading && <p className="animate__animated animate__flash">Loading...</p>
        }

        {
          gifs.map((gif) => (
            <GifGridItem 
              key={gif.id} 
              {...gif} 
            />
          ))
        }
      </div>
    </>
  )
}
