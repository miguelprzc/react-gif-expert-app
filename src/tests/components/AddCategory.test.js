import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('AddCategory', () => {
  const setCategories = jest.fn();
  const setInputValue = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  
  test('should render AddCategory properly', () => {
    
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should change the input box', () => {
    const input = wrapper.find('input');
    input.simulate('change', {target: {value: 'Hello world'}});
    
    //expect(setInputValue).toHaveBeenCalled();
  });
  
  test('should not post information on submit', () => {
    wrapper.find('form').simulate('submit', {preventDefault: () => {}});
    
    expect(setCategories).not.toHaveBeenCalled();
  });
  
  test('should call setCategories and clean input box', () => {
    const input = wrapper.find('input');
    input.simulate('change', {target: {value: 'Hello world'}});
    wrapper.find('form').simulate('submit', {preventDefault: () => {}});
    
    expect(setCategories).toHaveBeenCalled();
    expect(input.text()).toBe('');
  });
  
  
  
});