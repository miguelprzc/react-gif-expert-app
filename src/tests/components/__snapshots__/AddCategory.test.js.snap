// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should render AddCategory properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="What are you looking for?"
    type="text"
    value=""
  />
</form>
`;
